### PR TITLE
Configure dependabot + Upgrade netcoreapp 5.0

### PR DESCRIPTION
**Pull request by zengyu714**
*Issue #, if available:*

*Description of changes:*
- `dependabot` is configured to upgrade old dependencies
- Fix errors in sample application 
  - Allow producer to skip creating stream when exists rather than throw exception
  - Add user-agent to package downloader
  
*Testing*
`producer`
```
Putting records in stream : kclnetsample
Successfully putrecord 0:
	 partition key =  partitionKey-0, shard ID = shardId-000000000000
Successfully putrecord 1:
	 partition key =  partitionKey-1, shard ID = shardId-000000000000
Successfully putrecord 2:
	 partition key =  partitionKey-2, shard ID = shardId-000000000000
Successfully putrecord 3:
```
`consumer`
```
2021-12-29 16:55:35,704 [multi-lang-daemon-0001] ERROR s.a.k.multilang.DrainChildSTDERRTask [NONE] - Received error line from subprocess [	sequence number = 49625319226340669966826554256544095024018277912222367746,] for shard shardId-000000000000
	sequence number = 49625319226340669966826554256544095024018277912222367746,
2021-12-29 16:55:35,704 [multi-lang-daemon-0001] ERROR s.a.k.multilang.DrainChildSTDERRTask [NONE] - Received error line from subprocess [	data = testData-7] for shard shardId-000000000000
	data = testData-7
2021-12-29 16:55:35,705 [multi-lang-daemon-0001] ERROR s.a.k.multilang.DrainChildSTDERRTask [NONE] - Received error line from subprocess [Retrieved record:] for shard shardId-000000000000
Retrieved record:
2021-12-29 16:55:35,705 [multi-lang-daemon-0001] ERROR s.a.k.multilang.DrainChildSTDERRTask [NONE] - Received error line from subprocess [	partition key = partitionKey-8,] for shard shardId-000000000000
	partition key = partitionKey-8,
2021-12-29 16:55:35,705 [multi-lang-daemon-0001] ERROR s.a.k.multilang.DrainChildSTDERRTask [NONE] - Received error line from subprocess [	sequence number = 49625319226340669966826554256545303949837892610116550658,] for shard shardId-000000000000
	sequence number = 49625319226340669966826554256545303949837892610116550658,
2021-12-29 16:55:35,705 [multi-lang-daemon-0001] ERROR s.a.k.multilang.DrainChildSTDERRTask [NONE] - Received error line from subprocess [	data = testData-8] for shard shardId-000000000000
	data = testData-8
```
- Note the [log line](https://github.com/awslabs/amazon-kinesis-client-net/blob/master/SampleConsumer/SampleConsumer.cs#L88) is flushed at ERROR level

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: SampleProducer/SampleProducer.cs [hunk 1]
```diff
@@ -60,11 +60,17 @@ public static void Main(string[] args)
                 var CreateStreamResponse = kinesisClient.CreateStreamAsync(createStreamReq).Result;
                 Console.Error.WriteLine("Created Stream : " + myStreamName);
             }
-            catch (ResourceInUseException)
+            catch (Exception ex)
             {
-                Console.Error.WriteLine("Producer is quitting without creating stream " + myStreamName +
-                    " to put records into as a stream of the same name already exists.");
-                Environment.Exit(1);
+                if (ex is AggregateException || ex is ResourceInUseException)
```
Comment (QAQJ) [inline SampleProducer/SampleProducer.cs:65]: Under what circumstances will `AggregateException` happen?

Comment (zengyu714) [reply SampleProducer/SampleProducer.cs:65]: `AggregateException` is the parent exception of `ResourceInUseException`, yeah i think another level of if: `ex is AggregateException && ex is ResourceInUseException` is more appropriate!

Comment (zengyu714) [reply SampleProducer/SampleProducer.cs]: Hey, improved the error handling!

**Review by QAQJ — APPROVED** [review]
LGTM!

**Pull request merged**